docs/proposals/proposal_Updatable_VM_fields.md

```diff
@@ -0,0 +1,92 @@
+# Handling VM properties which DON'T trigger rolling update(specifically tags)
+
+## Current Situation with Tags
+
+We pass the `labels` which user provides in [shootYaml](https://github.com/gardener/gardener/blob/fb29d38e6615ed17d409a8271a285254d9dd00ad/example/90-shoot.yaml#L61-L62) as `tags` to the VM.
+For doing that we update the machineClass. This update is done by gardener-extension-provider-XYZ.
+Currently 
+- in aws provider directly passes the labels to the [tag section](https://github.com/gardener/gardener-extension-provider-aws/blob/0a740eeca301320275d77d1c48d3c32d4ebcd7dd/pkg/controller/worker/machines.go#L158-L164) of machineClass
+- in [azure](https://github.com/gardener/gardener-extension-provider-azure/blob/b6424f0122e174863e783555aa0ad68700edd87b/pkg/controller/worker/machines.go#L371-L373) sanitize them, and then pass. 
+- in [GCP](https://github.com/gardener/gardener-extension-provider-gcp/blob/eb851f716e45336b486f3aaf46268859de2adecb/pkg/controller/worker/machines.go#L312-L315) sanitize them, and then pass.
+
+Looking at an example of what machineClass tag section(from aws) looks like:
+```
+tags:
+    kubernetes.io/arch: amd64
+    networking.gardener.cloud/node-local-dns-enabled: "true"
+    node.kubernetes.io/role: node
+    worker.garden.sapcloud.io/group: worker-1
+    worker.gardener.cloud/cri-name: containerd
+    worker.gardener.cloud/pool: worker-1
+    worker.gardener.cloud/system-components: "true"
+    
+    kubernetes.io/cluster/shoot--i544024--rolling-test: "1"
+    kubernetes.io/role/node: "1"     
+                              
+    testlabel: "true"                                          
+```
+
+section1 -> [g/g](https://github.com/gardener/gardener/blob/c11c86ae07d8ea784f5c41362cd41800f06bb3ed/pkg/operation/botanist/component/extensions/worker/worker.go#L171-L197)</br>
+section2 -> [extension-provider](https://github.com/gardener/gardener-extension-provider-aws/blob/0a740eeca301320275d77d1c48d3c32d4ebcd7dd/pkg/controller/worker/machines.go#L160-L161)</br>
+section3 -> by user
+
+Out of these , MCM needs to put following tags(calling it `Must Tags`) on the VM for its orphan collection and GetVMStatus logic to work, can be seen in section2 above:
```

Comment:
MCM needs to consume and not put as these tags are put by `gardener-extension-<provider>`